matdeeplearn/preprocessor/helpers.py

```diff
@@ -1,14 +1,29 @@
-import numpy as np
-import ase
-from ase import io
-import torch
+import contextlib
 import itertools
 from pathlib import Path
 
+import ase
+import numpy as np
 import torch
 import torch.nn.functional as F
-from torch_geometric.utils import dense_to_sparse, degree, add_self_loops
+from ase import io
+from torch.profiler import ProfilerActivity, profile, record_function
 from torch_geometric.data.data import Data
+from torch_geometric.utils import add_self_loops, degree, dense_to_sparse
+from torch_sparse import SparseTensor
+
+
+@contextlib.contextmanager
+def prof_ctx():
+    """Primitive debug tool which allows profiling of PyTorch code"""
+    with profile(
+        activities=[ProfilerActivity.CUDA], record_shapes=True, profile_memory=True
+    ) as prof:
+
+        yield
+
+    print(prof.key_averages().table(sort_by="cuda_memory_usage", row_limit=10))
```

Comment:
Should we switch this to logging too, or does print make more sense here?

Reply:
I think debug would make the most sense as it's a debug tool. For future development, there are better tracing tools for PyTorch such as one provided by W&B that we could look into.